From 3e6d0989db94335ab3e59f7bfa5fed9b877a9be1 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 4 Jan 2011 17:46:40 -0200
Subject: [RHEL6 qemu-kvm PATCH 1/2] Revert "spice/qxl: make draw_area and vgafb share memory"

RH-Reverts: 211fb5b4dce32fd82508133e12d9d13098310cdb
RH-Reverts-patchwork-id: 12134
Bugzilla-related: 653334
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/qxl.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 0db910b..d6c516b 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -733,21 +733,22 @@ static void init_qxl_ram(PCIQXLDevice *d, uint8_t *buf, uint32_t actual_ram_size
     *item = 0;
 
     if (d->id == 0) {
-        if (draw_area_size < VGA_RAM_SIZE) {
-            draw_area_size = d->shadow_rom.draw_area_size = VGA_RAM_SIZE;
-        }
+        d->shadow_rom.draw_area_offset = VGA_RAM_SIZE;
+    } else {
+        d->shadow_rom.draw_area_offset = 0;
     }
-    d->shadow_rom.draw_area_offset = 0;
     d->shadow_rom.pages_offset = d->shadow_rom.draw_area_offset + draw_area_size;
     d->shadow_rom.num_io_pages = (actual_ram_size - ram_header_size - d->shadow_rom.pages_offset) >> TARGET_PAGE_BITS;
 
     *d->rom = d->shadow_rom;
 
     dprintf(1, "qxl device memory layout (device #%d)\n"
-            "  vga/draw:   0x%08x\n"
+            "  vga ram:    0x%08x\n"
+            "  draw area:  0x%08x\n"
             "  io pages:   0x%08x (%d pages)\n"
             "  ram header: 0x%08x\n",
             d->id, 0,
+            d->shadow_rom.draw_area_offset,
             d->shadow_rom.pages_offset, d->shadow_rom.num_io_pages,
             d->shadow_rom.ram_header_offset);
 }
-- 
1.7.3.2

